x-pack/plugins/alerting/server/task_runner/execution_handler.test.ts

```diff
@@ -729,4 +731,251 @@ describe('Execution Handler', () => {
+            "ruleId": "1",
+            "spaceId": "test1",
+          },
+        ]
+      `);
+    });
+
+    it('does not populate the rule.url when the base url is not specified', async () => {
+      const execParams = {
+        ...defaultExecutionParams,
+        rule: ruleWithUrl,
+        taskRunnerContext: {
+          ...defaultExecutionParams.taskRunnerContext,
+          kibanaBaseUrl: undefined,
+        },
+      };
+
+      const executionHandler = new ExecutionHandler(generateExecutionParams(execParams));
+      await executionHandler.run(generateAlert({ id: 1 }));
+
+      expect(injectActionParamsMock.mock.calls[0]).toMatchInlineSnapshot(`
+        Array [
+          Object {
+            "actionParams": Object {
+              "val": "rule url: ",
+            },
+            "actionTypeId": "test",
+            "ruleId": "1",
+            "spaceId": "test1",
+          },
+        ]
+      `);
+    });
+
+    it('does not populate the rule.url when the alert id is not defined', async () => {
+      const execParams = {
+        ...defaultExecutionParams,
+        rule: ruleWithUrl,
+        taskRunnerContext: {
+          ...defaultExecutionParams.taskRunnerContext,
+          kibanaBaseUrl: 'http://localhost:12345',
+        },
+        taskInstance: {
+          params: { spaceId: 'test1' },
+        } as unknown as ConcreteTaskInstance,
+      };
+
+      const executionHandler = new ExecutionHandler(generateExecutionParams(execParams));
+      await executionHandler.run(generateAlert({ id: 1 }));
+
+      expect(injectActionParamsMock.mock.calls[0]).toMatchInlineSnapshot(`
+        Array [
+          Object {
+            "actionParams": Object {
+              "val": "rule url: ",
+            },
+            "actionTypeId": "test",
+            "ruleId": undefined,
+            "spaceId": "test1",
+          },
+        ]
+      `);
+    });
+
+    it('does not populate the rule.url when the alert id is set to undefined', async () => {
+      const execParams = {
+        ...defaultExecutionParams,
+        rule: ruleWithUrl,
+        taskRunnerContext: {
+          ...defaultExecutionParams.taskRunnerContext,
+          kibanaBaseUrl: 'http://localhost:12345',
+        },
+        taskInstance: {
+          params: { spaceId: 'test1', alertId: undefined },
+        } as unknown as ConcreteTaskInstance,
+      };
+
+      const executionHandler = new ExecutionHandler(generateExecutionParams(execParams));
+      await executionHandler.run(generateAlert({ id: 1 }));
+
+      expect(injectActionParamsMock.mock.calls[0]).toMatchInlineSnapshot(`
+        Array [
+          Object {
+            "actionParams": Object {
+              "val": "rule url: ",
+            },
+            "actionTypeId": "test",
+            "ruleId": undefined,
+            "spaceId": "test1",
+          },
+        ]
+      `);
+    });
+
+    it('does not populate the rule.url when the alert id is a number', async () => {
```

Comment:
Yeah the reason I thought this was worth covering is because the way we've typed the task manager params is `Record<string, any>`. So when we destructure `taskInstance.params` it's showing `alertId: ruleId` as type `any`. So I figured it's probably worth skipping the rule building if the type is anything but a string 🤷‍♂️ 

![image](https://user-images.githubusercontent.com/56361221/201775843-5d608307-66ff-4bdd-a86b-dc32febb924b.png)

Reply:
Ah!  Looking closer, I missed the `buildRuleUrl()` method is referencing `this.taskInstance.params.alertId` - I missed that before, and that does seem a little brittle.  Looking at it's call site, it looks like you could instead pass `ruleId` as a parameter.

Reply:
Ah, good catch - ya, that variable is extracted from the same place 🤦🏻‍♂️ .

However, I noticed a `this.rule` available, so I think you can leave it parameterless and change

`this.taskInstance.params.alertId` -> `this.rule.id`